R/utils.R

```diff
@@ -82,23 +82,25 @@ include_js_files <- function(pattern = "*") { # nolint
 #' @param title (`character(1)`) title of the card (unless overwritten by label)
 #' @param label (`character(1)`) label provided by the user when adding the card
 #' @param description (`character(1)`) optional additional description
-#' @param with_filter (`logical(1)`) flag indicating to add filter state
 #' @param filter_panel_api (`FilterPanelAPI`) object with API that allows the generation
 #' of the filter state in the report
 #'
 #' @return (`TealReportCard`) populated with a title, description and filter state
 #'
 #' @keywords internal
-card_template <- function(title, label, description = NULL, with_filter, filter_panel_api) {
+card_template <- function(title, label, description = NULL, filter_panel_api) {
+  checkmate::assert_string(title)
```

Comment:
sorry can we please take out the `checkmate::` name space thing since we import the whole of it?

Reply:
putting package name in front of package function improves the process of debugging if you debug the function interactively

Reply:
teal is also imported as a whole but we do put :: in front of teal function names https://github.com/search?q=repo%3Ainsightsengineering%2Fteal.modules.hermes%20%3A%3A&type=code

https://github.com/insightsengineering/teal.modules.hermes/blob/3d16439ea447c4a0aee910270aeab6ebe9dc61de/R/scatterplot.R#L48

Reply:
yeah we should also take out the `teal::` bits please. thanks!

Reply:
@kartikeyakirar have you seen this conversation about `teal`

Reply:
Okay, I will take out teal:: call as well.

Reply:
Here is task : https://github.com/insightsengineering/teal.modules.hermes/issues/337